src/typescript-service.ts

```diff
@@ -1358,3 +1401,26 @@ function pushall<T>(arr: T[], ...elems: (T | null | undefined)[]): number {
 	}
 	return arr.length;
 }
+
+/**
+ * Maps string-based CompletionEntry::kind to enum-based CompletionItemKind
+ */
+const completionKinds: { [name: string]: CompletionItemKind } = {
```

Comment:
Interesting, didn't know about this, thanks for the link!
Would be nice if the Microsoft used `preserveConstEnums` for the package.